File: Two Sum/Two Sum.py
class Solution:
    # @return a tuple, (index1, index2)
    def twoSum(self, num, target):
        mymap = dict ()
        ans = []
        for i in range (len (num)) :
            if mymap.has_key (target - num[i]) :
                ans.append (mymap.get (target - num[i]))
                ans.append (i + 1)
                return ans
            mymap[num[i]] = i + 1
        return ans

File: Two Sum/Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int> &numbers, int target) {
        map <int , int> mymap;
        vector <int> ans;
        for (int i = 0 ; i < numbers.size () ; i ++) {
            if (mymap.find (target - numbers[i]) != mymap.end ()) {
                ans.push_back (mymap[target - numbers[i]]);
                ans.push_back (i + 1);
                return ans;
            }
            mymap[numbers[i]] = i + 1;
        }
    }
};